__tests__/restore.test.ts

```diff
@@ -205,3 +205,88 @@ test("restore with cache found for restore key", async () => {
     );
     expect(failedMock).toHaveBeenCalledTimes(0);
 });
+
+test("Fail restore when fail on cache miss is enabled and primary + restore keys not found", async () => {
+    const path = "node_modules";
+    const key = "node-test";
+    const restoreKey = "node-";
+    testUtils.setInputs({
+        path: path,
+        key,
+        restoreKeys: [restoreKey],
+        failOnCacheMiss: true
```

Comment:
👍🏻 Done. Additionally, since I used `utils.getInputAsBool(Inputs.FailOnCacheMiss)` to get the input, basically all tests where `failOnCacheMiss` isn't explicitly set to `true` will test any remaining cases.

https://github.com/actions/cache/blob/8e3048d0f72c72babe53747a8f10e16a74c19ba1/src/utils/actionUtils.ts#L55-L60

Reply:
Great thanks :)